crates/re_data_source/src/data_loader/loader_archetype.rs

```diff
@@ -0,0 +1,157 @@
+        tx: std::sync::mpsc::Sender<LoadedData>,
+    ) -> Result<(), crate::DataLoaderError> {
+        use anyhow::Context as _;
+
+        if filepath.is_dir() {
+            return Ok(()); // simply not interested
+        }
+
+        re_tracing::profile_function!(filepath.display().to_string());
+
+        let contents = std::fs::read(&filepath)
+            .with_context(|| format!("Failed to read file {filepath:?}"))?;
+        let contents = std::borrow::Cow::Owned(contents);
+
+        self.load_from_file_contents(store_id, filepath, contents, tx)
+    }
+
+    fn load_from_file_contents(
+        &self,
+        _store_id: re_log_types::StoreId,
+        filepath: std::path::PathBuf,
+        contents: std::borrow::Cow<'_, [u8]>,
+        tx: std::sync::mpsc::Sender<LoadedData>,
+    ) -> Result<(), crate::DataLoaderError> {
+        re_tracing::profile_function!(filepath.display().to_string());
+
+        let entity_path = EntityPath::from_file_path(&filepath);
+
+        let timepoint = TimePoint::timeless();
+        // TODO: If I log these times, the space views disappear and cannot be created manually
+        // if let Ok(metadata) = filepath.metadata() {
+        //     use re_log_types::{Time, Timeline};
+        //
+        //     if let Some(created) = metadata.created().ok().and_then(|t| Time::try_from(t).ok()) {
+        //         timepoint.insert(Timeline::new_temporal("created_at"), created.into());
+        //     }
+        //     if let Some(modified) = metadata
+        //         .modified()
+        //         .ok()
+        //         .and_then(|t| Time::try_from(t).ok())
+        //     {
+        //         timepoint.insert(Timeline::new_temporal("modified_at"), modified.into());
+        //     }
+        //     if let Some(accessed) = metadata
+        //         .accessed()
+        //         .ok()
+        //         .and_then(|t| Time::try_from(t).ok())
+        //     {
+        //         timepoint.insert(Timeline::new_temporal("accessed_at"), accessed.into());
+        //     }
+        // }
+
+        let extension = filepath
+            .extension()
+            .unwrap_or_default()
+            .to_ascii_lowercase()
+            .to_string_lossy()
+            .to_string();
+
+        let mut rows = Vec::new();
+
+        if crate::SUPPORTED_MESH_EXTENSIONS.contains(&extension.as_str()) {
+            re_log::debug!(?filepath, loader = self.name(), "Loading 3D model…",);
+            rows.extend(load_mesh(
+                filepath,
+                timepoint,
+                entity_path,
+                contents.into_owned(),
+            )?);
+        } else if crate::SUPPORTED_IMAGE_EXTENSIONS.contains(&extension.as_str()) {
+            re_log::debug!(?filepath, loader = self.name(), "Loading image…",);
+            rows.extend(load_image(
+                &filepath,
+                timepoint,
+                entity_path,
+                contents.into_owned(),
+            )?);
+        };
```

Comment:
shouldn't we have an `else` here with `crate::DataLoaderError::UnsupportedExtension` or something?

Reply:
Hmmmm I guess we could if there are no external loaders registered :thinking:

Reply:
I'll do that as part of the "unsupported support" follow up PR